src/backend/main/py.main/caliopen_main/parsers/mail.py

```diff
@@ -132,19 +206,21 @@ def __init__(self, raw_data):
             # XXX what to do ?
             log.warn('Defects on parsed mail %r' % self.mail.defects)
             self.warning = self.mail.defects
+        self.get_bodies()
 
     @property
     def subject(self):
         """Mail subject."""
         return self.mail.get('Subject')
 
     @property
-    def body(self):
-        """Mail body."""
-        # XXX define the extraction logic for multipart and co.
-        if not self.mail.is_multipart():
-            return self.mail.get_payload()
-        return ''
+    def preferred_body(self):
```

Comment:
body is defined by the zope interface implemented by the MailParser. You can't change it this way without correct update of the interface (you remove body and add 2 others attributes that I don't see implemented by this parser)